components/style/properties/longhand/box.mako.rs

```diff
@@ -1144,39 +1145,453 @@ ${helpers.predefined_type("scroll-snap-coordinate",
     }
 
     fn parse_two_angles(context: &ParserContext, input: &mut Parser)
-                       -> Result<(specified::Angle, specified::Angle),()> {
-        let first = try!(specified::Angle::parse(context, input));
+                       -> Result<(SpecifiedAngle, SpecifiedAngle),()> {
+        let first = try!(SpecifiedAngle::parse(context, input));
         let second = input.try(|input| {
             try!(input.expect_comma());
-            specified::Angle::parse(context, input)
-        }).unwrap_or(specified::Angle(0.0));
+            SpecifiedAngle::parse(context, input)
+        }).unwrap_or(SpecifiedAngle::zero());
         Ok((first, second))
     }
 
+    // https://drafts.csswg.org/css-transforms/#mathematical-description
+    // Note that the matrix elements are addressed in column-major order.
+    #[derive(Clone, Copy, Debug, PartialEq)]
+    #[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+    pub enum SpecifiedMatrix {
+        Matrix(CSSFloat, CSSFloat, CSSFloat, CSSFloat, CSSFloat, CSSFloat),
+        Matrix3D(self::computed_value::ComputedMatrix),
+    }
+
+    impl fmt::Display for SpecifiedMatrix {
+        fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+            match *self {
+                SpecifiedMatrix::Matrix(a, b, c, d, e, f) => {
+                    try!(formatter.write_str("matrix("));
+                        try!(formatter.write_str(a.to_string().as_str()));
+                        try!(formatter.write_str(", "));
+                        try!(formatter.write_str(b.to_string().as_str()));
+                        try!(formatter.write_str(", "));
+                        try!(formatter.write_str(c.to_string().as_str()));
+                        try!(formatter.write_str(", "));
+                        try!(formatter.write_str(d.to_string().as_str()));
+                        try!(formatter.write_str(", "));
+                        try!(formatter.write_str(e.to_string().as_str()));
+                        try!(formatter.write_str(", "));
+                        try!(formatter.write_str(f.to_string().as_str()));
```

Comment:
This (and many places below) shouldn't have an extra level of indent. The editor may think the `(` inside the string opens a level, but that is not true. Intend should only follow the levels in the source code.